cpp/roaring.hh

```diff
@@ -11,6 +11,14 @@ A C++ header for Roaring Bitmaps.
 #include <stdexcept>
 #include <string>
 
+#ifndef ROARING_EXCEPTIONS
+#if __cpp_exceptions
+#define ROARING_EXCEPTIONS 1
+#else
+#define ROARING_EXCEPTIONS 0
+#endif
+#endif
```

Comment:
Another option that might be more flexible would be to define a preprocessor macro describing how to handle exceptions.  The macro could take a single argument and the default with exceptions would be to throw with that argument, and without exceptions the default would be to terminate ignoring the argument.

Then users could define their own macro before including the header that did other things such as call their own throw with different/extra arguments, or their own special terminate, or something else.  The requirement would have to be that the function never returns of course.  You could play some tricks to get the compiler to warn if that wasn't true but may not be worth it.

Reply:
Would you  be willing to produce a pull request that would improve in my approach?

Reply:
I can do that

Reply:
Great. So I will freeze this PR and give you a chance to improve it (or produce your own).